bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/dto/DataPayload.java

```diff
@@ -168,174 +145,202 @@ public double getTotalEnergyExportT1Kwh() {
-     * @return total imported gas volume
+     * @return current active total power
      */
-    public double getTotalGasM3() {
-        return totalGasM3;
+    public int getActivePowerW() {
+        return activePowerW;
     }
 
     /**
-     * Setter for the total imported gas volume
+     * Getter for the current active total power on phase 2
      *
-     * @param totalGasM3 total imported gas volume
+     * @return current active total power on phase 2
      */
-    public void setTotalGasM3(double totalGasM3) {
-        this.totalGasM3 = totalGasM3;
+    public int getActivePowerL1W() {
+        return activePowerL1W;
     }
 
     /**
-     * Getter for the time stamp of the last gas update
+     * Getter for the current active total power on phase 2
      *
-     * @return time stamp of the last gas update
+     * @return current active total power on phase 2
      */
-    public long getGasTimestamp() {
-        return gasTimestamp;
+    public int getActivePowerL2W() {
+        return activePowerL2W;
     }
 
     /**
-     * Setter for the time stamp of the last gas update
+     * Getter for the current active total power on phase 3
      *
-     * @param gasTimestamp time stamp of the last gas update
+     * @return current active total power on phase 3
      */
-    public void setGasTimestamp(long gasTimestamp) {
-        this.gasTimestamp = gasTimestamp;
+    public int getActivePowerL3W() {
+        return activePowerL3W;
     }
 
     /**
-     * Getter for the total imported water volume
+     * Getter for the total imported gas volume
      *
-     * @return total imported water volume
+     * @return total imported gas volume
      */
-    public double getTotalWaterM3() {
-        return totalWaterM3;
+    public double getTotalGasM3() {
+        return totalGasM3;
     }
 
     /**
-     * Setter for the total imported water volume
+     * Getter for the time stamp of the last gas update
      *
-     * @param totalWaterM3 total imported water volume
+     * @return time stamp of the last gas update as ZonedDateTime
      */
-    public void setTotalWaterM3(double totalWaterM3) {
-        this.totalWaterM3 = totalWaterM3;
+    public @Nullable ZonedDateTime getGasTimestamp() {
+        long dtv = gasTimestamp;
+        if (dtv < 1) {
+            return null;
+        }
+
+        // 210119164000
+        int seconds = (int) (dtv % 100);
+
+        dtv /= 100;
+        int minutes = (int) (dtv % 100);
+
+        dtv /= 100;
+        int hours = (int) (dtv % 100);
+
+        dtv /= 100;
+        int day = (int) (dtv % 100);
+
+        dtv /= 100;
+        int month = (int) (dtv % 100);
+
+        dtv /= 100;
+        int year = (int) (dtv + 2000);
+
+        try {
+            return ZonedDateTime.of(year, month, day, hours, minutes, seconds, 0, ZoneId.systemDefault());
+        } catch (DateTimeException e) {
+            LoggerFactory.getLogger(DataPayload.class).warn("Unable to parse Gas timestamp: {}", gasTimestamp);
```

Comment:
I'm not sure we should log from a DTO get method. Perhaps this could be logged in the handler instead (even if this would require a getter for the raw value to be logged)?